classes/ai_access_handler.php

```diff
@@ -0,0 +1,140 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace quizaccess_ai;
+
+use local_ai_manager\ai_manager_utils;
+use stdClass;
+
+/**
+ * Checks and fetches the AI availability state for quizaccess_ai.
+ *
+ * @package   quizaccess_ai
+ * @copyright 2025 ISB Bayern
+ * @author    Thomas Schönlein
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class ai_access_handler {
+    /**
+     * Checks whether the AI features required for this quiz are available.
+     *
+     * @param stdClass $user The user for whom the availability should be checked.
+     * @param int $contextid The context id of the quiz.
+     * @param array|null $requiredpurposes Purposes that must be enabled. If null, defaults are used.
+     * @return bool|string True when AI is available, error message otherwise.
+     */
+    public function is_available(stdClass $user, int $contextid, ?array $requiredpurposes = null): bool|string {
+        $requiredpurposes = $requiredpurposes ?? $this->get_required_purposes();
```

Comment:
Also inline comment that for now only qtype_aitext in combination with local_ai_manager is being checked.